File: problems/majority_elements/MajorityElement.java
package problems.majority_elements;

public class MajorityElement {
    public int majorityElement(int[] nums) {
        int majorElement = nums[0];
        int count = 1;

        for (int i = 1; i < nums.length; i++) {
            if (majorElement == nums[i]) {
                count += 1;
            } else {
                count -= 1;
                if (count == 0) {
                    majorElement = nums[i];
                    count = 1;
                }
            }
        }
        return majorElement;
    }
}

File: problems/majority_elements/majority_element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        major = nums[0]
        count = 1
        for num in nums[1:]:
            if num == major:
                count += 1
            else:
                count -= 1
                if count == 0:
                    major = num
                    count = 1
        return major
